data-prepper-api/src/main/java/org/opensearch/dataprepper/model/source/Source.java

```diff
@@ -25,4 +25,11 @@
      * Notifies the source to stop consuming/writing records into Buffer.
      */
     void stop();
+
+    /**
+     * Indicates if the source has end to end acknowledgements enabled
+     *
+     * @return boolean indicating if the source enabled end-to-end acknowledgements
+     */
+    boolean enabledAcknowledgements();
```

Comment:
Please create this as a default interface method.

```
default boolean enabledAcknowledgements() {
  return false;
}
```

You don't even have to implement this method in the classes which do not support acknowledgements.

Also, Java conventions are normally to make a boolean method more like a question. So this. could be `areAcknowledgementsEnabled`. I don't feel very strongly on this point though.